compiler/aarch64/codegen/OMRMachine.cpp

```diff
@@ -771,16 +770,14 @@ void OMR::ARM64::Machine::coerceRegisterAssignment(TR::Instruction *currentInstr
       TR::RealRegister *spareReg = NULL;
       TR::Register *currentTargetVirtual = targetRegister->getAssignedRegister();
 
-      bool needTemp = (rk == TR_FPR || rk == TR_VRF); // xor is unavailable for register exchange
-
       if (targetRegister->getState() == TR::RealRegister::Blocked)
          {
 #ifdef DEBUG
          if (comp->getOption(TR_TraceCG))
             diagnostic(", which is blocked and assigned to %s",
                        currentTargetVirtual->getRegisterName(comp));
 #endif
-         if (!currentAssignedRegister || needTemp)
+         if (!currentAssignedRegister)
```

Comment:
Please merge the `if` block from L.781 to L.791 with the `else` block from L.803 to L.823.

Reply:
Merged the blocks.